azurerm/resource_arm_storage_blob.go

```diff
@@ -101,686 +91,228 @@ func resourceArmStorageBlob() *schema.Resource {
-			container: container,
-			name:      name,
-			blocks:    blocks,
-			errors:    errors,
-			wg:        wg,
-			attempts:  attempts,
-		})
-	}
-
-	wg.Wait()
-
-	if len(errors) > 0 {
-		return fmt.Errorf("Error while uploading source file %q: %s", source, <-errors)
-	}
-
-	containerReference := client.GetContainerReference(container)
-	blobReference := containerReference.GetBlobReference(name)
-	blobReference.Properties.ContentType = contentType
-	options := &storage.PutBlockListOptions{}
-	err = blobReference.PutBlockList(blockList, options)
-	if err != nil {
-		return fmt.Errorf("Error updating block list for source file %q: %s", source, err)
-	}
-
-	return nil
-}
-
-func resourceArmStorageBlobBlockSplit(file *os.File) ([]storage.Block, []resourceArmStorageBlobBlock, error) {
-	const (
-		idSize          = 64
-		blockSize int64 = 4 * 1024 * 1024
-	)
-	var parts []resourceArmStorageBlobBlock
-	var blockList []storage.Block
-
-	info, err := file.Stat()
-	if err != nil {
-		return nil, nil, fmt.Errorf("Error stating source file %q: %s", file.Name(), err)
-	}
-
-	for i := int64(0); i < info.Size(); i = i + blockSize {
-		entropy := make([]byte, idSize)
-		_, err = rand.Read(entropy)
-		if err != nil {
-			return nil, nil, fmt.Errorf("Error generating a random block ID for source file %q: %s", file.Name(), err)
-		}
-
-		sectionSize := blockSize
-		remainder := info.Size() - i
-		if remainder < blockSize {
-			sectionSize = remainder
-		}
-
-		block := storage.Block{
-			ID:     base64.StdEncoding.EncodeToString(entropy),
-			Status: storage.BlockStatusUncommitted,
-		}
-
-		blockList = append(blockList, block)
-
-		parts = append(parts, resourceArmStorageBlobBlock{
-			id:      block.ID,
-			section: io.NewSectionReader(file, i, sectionSize),
-		})
-	}
-
-	return blockList, parts, nil
-}
-
-type resourceArmStorageBlobBlockUploadContext struct {
-	client    *storage.BlobStorageClient
-	container string
-	name      string
-	source    string
-	attempts  int
-	blocks    chan resourceArmStorageBlobBlock
-	errors    chan error
-	wg        *sync.WaitGroup
-}
-
-func resourceArmStorageBlobBlockUploadWorker(ctx resourceArmStorageBlobBlockUploadContext) {
-	for block := range ctx.blocks {
-		buffer := make([]byte, block.section.Size())
-
-		_, err := block.section.Read(buffer)
-		if err != nil {
-			ctx.errors <- fmt.Errorf("Error reading source file %q: %s", ctx.source, err)
-			ctx.wg.Done()
-			continue
-		}
-
-		for i := 0; i < ctx.attempts; i++ {
-			container := ctx.client.GetContainerReference(ctx.container)
-			blob := container.GetBlobReference(ctx.name)
-			options := &storage.PutBlockOptions{}
-			if err = blob.PutBlock(block.id, buffer, options); err == nil {
-				break
-			}
-		}
-		if err != nil {
-			ctx.errors <- fmt.Errorf("Error uploading block %q for source file %q: %s", block.id, ctx.source, err)
-			ctx.wg.Done()
-			continue
-		}
+	d.SetId(id)
 
-		ctx.wg.Done()
-	}
+	return resourceArmStorageBlobUpdate(d, meta)
```

Comment:
(for info) this now calls into Update because the Access Tier might as well be set there, since it needs to be configured after creation anyway